samples/quantum-approximation-optimization/qaoa-for-quadratic-unconstrained-binary-optimization.ipynb

```diff
@@ -0,0 +1,820 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": "The Rydberg blockade radius is directly linked to the Rabi frequency $\\Omega$ and is obtained using `DigitalAnalogDevice.rydberg_blockade_radius()`. In this notebook, $\\Omega$ is initially fixed to a frequency of 1 rad/µs. We can therefore build the adjacency matrix $A$ of $G$ in the following way:"
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To ensure that we are not exciting the system to states that are too excited, we keep $\\Omega \\in [0, \\Omega_{\\text{max}}]$, and choose $\\Omega_{\\text{max}}$ as the median of the values of Q to ensures that the adiabatic path is efficient."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "# We choose a median value between the min and the max\n",
+    "Omega = np.median(Q[Q > 0].flatten())\n",
+    "delta_0 = -5  # just has to be negative\n",
+    "delta_f = -delta_0  # just has to be positive\n",
+    "T = 4000  # time in ns, we choose a time long enough to ensure the propagation of information in the system"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "adiabatic_pulse = Pulse(\n",
+    "    InterpolatedWaveform(T, [1e-9, Omega, 1e-9]),\n",
+    "    InterpolatedWaveform(T, [delta_0, 0, delta_f]),\n",
+    "    0,\n",
+    ")\n",
+    "seq = Sequence(reg, DigitalAnalogDevice)\n",
+    "seq.declare_channel(\"ising\", \"rydberg_global\")\n",
+    "seq.add(adiabatic_pulse, \"ising\")\n",
+    "seq.draw()"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "simul = QutipEmulator.from_sequence(seq)\n",
+    "results = simul.run()\n",
+    "final = results.get_final_state()\n",
+    "count_dict = results.sample_final_state()\n",
+    "print(results.sample_final_state())"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "plot_distribution(count_dict)"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "See how fast and performant this method is! In only a few micro-seconds, we find an excellent solution."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How does the time evolution affect the quality of the results?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "cost_qutip_emulator = []\n",
+    "for T in 1000 * np.linspace(1, 10, 10):\n",
+    "    seq = Sequence(reg, DigitalAnalogDevice)\n",
```

Comment:
The demonstration would be better with AnalogDevice, but the value of T might have to be restrained to `np.linspace(1000, 5000, 10)` (I think the max duration of the Rydberg channel in AnalogDevice is 5000ns)

Reply:
So I'm not confident about taking the risk of breaking the notebook, is it ok to leave as it is ?

Reply:
It's okay to leave it as it is